nf_core/modules/module_test.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""
+The ModulesTest class runs the tests locally
+"""
+
+import logging
+import questionary
+import os
+import pytest
+import sys
+import rich
+
+import nf_core.utils
+from .modules_repo import ModulesRepo
+
+log = logging.getLogger(__name__)
+
+class ModulesTest(object):
+    def __init__(
+        self,
+        module_name=None,
+        no_prompts=False,
+        pytest_args="",
+    ):
+        self.module_name = module_name
+        self.no_prompts = no_prompts
+        self.pytest_args = pytest_args
+
+    def run(self):
+        """Run test steps"""
+        if not self.no_prompts:
+            log.info(
+                "[yellow]Press enter to use default values [cyan bold](shown in brackets) [yellow]or type your own responses"
+            )
+        self.check_inputs()
+        self.set_profile()
+        self.run_pytests()
+
+    def check_inputs(self):
```

Comment:
I think this method as well as set_profiles and run_pytests could be private. If I understand the use case right they are intended for internal use only.